templates/oauth2/authorize.html

```diff
@@ -213,13 +220,22 @@
 
     <p>Please confirm that the application <strong>${client.description}</strong>
       (by <a href="/~${LOGIN(client.owner.profile.username)}">${client.owner.profile.username}</a>)
-      should be authorized to make requests on your behalf.</p>
+      should be authorized to make requests on your behalf with the following permissions:</p>
+
+    $if scopes:
+      <table id="permissions">
+        <tr><td><strong>PERMISSION</strong></td><td><strong>GRANTS</strong></td></tr>
+        $for scope in scopes:
+          <tr><td>${scope['name']}</td><td>${scope['description']}</td></tr>
+      </table>
+    $else:
+      <p>No elevated permissions requested</p>
```

Comment:
After carefully scouring the [documentation](http://tools.ietf.org/html/rfc6749#section-3.3) it appears that having an application with no scopes is not a valid configuration, so this can just be dropped